File: Codeforces/py/2059A.py
def can_get_three_distinct_sums(n, a, b):
    a.sort() 
    distinct_sums = set()
    
    for i in range(n):
        distinct_sums.add(a[i] + b[i])
        if len(distinct_sums) >= 3:
            return "YES"
    
    return "NO"


def solve():
    t = int(input().strip())
    
    for _ in range(t):
        n = int(input().strip())
        a = list(map(int, input().strip().split()))
        b = list(map(int, input().strip().split()))
        
        print(can_get_three_distinct_sums(n, a, b))

solve()

File: Codeforces/cpp solved/2059A.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

string can_get_three_distinct_sums(int n, vector<int>& a, vector<int>& b) {
    sort(a.begin(), a.end());
    set<int> distinct_sums;
    
    for (int i = 0; i < n; i++) {
        distinct_sums.insert(a[i] + b[i]);
        if (distinct_sums.size() >= 3) {
            return "YES";
        }
    }
    
    return "NO";
}

void solve() {
    int t;
    cin >> t;
    
    while (t--) {
        int n;
        cin >> n;
        vector<int> a(n), b(n);
        
        for (int i = 0; i < n; i++) cin >> a[i];
        for (int i = 0; i < n; i++) cin >> b[i];
        
        cout << can_get_three_distinct_sums(n, a, b) << endl;
    }
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    solve();
    return 0;
}
